gui/components/upload_file.py

```diff
@@ -16,12 +18,14 @@ def __upload_file(
         self, send_file_handler: Callable[[FileInfo], any], uploaded_file: UploadedFile
     ):
         if uploaded_file:
-            self.file_bytes: bytes = uploaded_file.getvalue()
-            file_info = FileInfo(uploaded_file.name, self.file_bytes)
-            self.upload_response: UploadVideoResponse = send_file_handler(file_info)
-            self.file_format = file_info.format
+            #TODO: Review serialization pipeline
+            json_data = json_celery_serializer(uploaded_file)
+            self.upload_response: UploadVideoResponse = process_file_task.delay(json_data)
+            
+            self.file_bytes = uploaded_file.getvalue()
+            self.file_format = os.path.basename(uploaded_file.name).split(".")[-1]
```

Comment:
Serialization pipeline is not optimized

Reply:
Na linha 26, pode ser self.file_format = file_info.format, pois o format já é calculado no construtor da classe FileInfo.